usersession/client/client_test.go

```diff
@@ -457,3 +458,18 @@ func (s *clientSuite) TestPendingRefreshNotification(c *C) {
 	err := s.cli.PendingRefreshNotification(context.Background(), &client.PendingSnapRefreshInfo{})
 	c.Assert(err, IsNil)
 }
+
+func (s *clientSuite) TestCloseRefreshNotification(c *C) {
+	s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		c.Assert(r.URL.Path, Equals, "/v1/notifications/close")
+		body, err := ioutil.ReadAll(r.Body)
+		c.Check(err, IsNil)
+		c.Check(string(body), DeepEquals, `{"instance-name":"some-snap"}`)
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(200)
+		w.Write([]byte(`{"type": "sync"}`))
+	})
+	err := s.cli.CloseRefreshNotification(context.Background(), &client.FinishedSnapRefreshInfo{InstanceName: "some-snap"})
+	c.Assert(err, IsNil)
```

Comment:
Maybe add a `n` counter to see that the handler is only called only once? Something like:
```
diff --git a/usersession/client/client_test.go b/usersession/client/client_test.go
index 82afd62406..bdb27ded65 100644
--- a/usersession/client/client_test.go
+++ b/usersession/client/client_test.go
@@ -460,7 +460,9 @@ func (s *clientSuite) TestPendingRefreshNotification(c *C) {
 }
 
 func (s *clientSuite) TestCloseRefreshNotification(c *C) {
+       n := 0
        s.handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+               n++
                c.Assert(r.URL.Path, Equals, "/v1/notifications/close")
                body, err := ioutil.ReadAll(r.Body)
                c.Check(err, IsNil)
@@ -472,4 +474,5 @@ func (s *clientSuite) TestCloseRefreshNotification(c *C) {
        })
        err := s.cli.CloseRefreshNotification(context.Background(), &client.FinishedSnapRefreshInfo{InstanceName: "some-snap"})
        c.Assert(err, IsNil)
+       c.Check(n, Equals, 1)
 }
```

Reply:
Thanks for adding this. Looks like my assumption that it's only called once is wrong:
```
FAIL: client_test.go:462: clientSuite.TestCloseRefreshNotification

client_test.go:477:
    c.Check(n, Equals, 1)
... obtained int = 2
... expected int = 1
```

Reply:
(unrelated to the PR) the test just before this one could also get this counter but that is really deep nitpick territory

Reply:
Ah, of course, that's because the test simulates 2 user sessions, and client sends the request to all client sockets. Silly me, I run go test in the wrong directory and missed that.